gef.py

```diff
@@ -4635,6 +4653,22 @@ def has_setting(self, name):
         key = self.__get_setting_name(name)
         return key in __config__
 
+    def update_setting(self, name, value, description=None):
+        """
+        Function to allow for internal commands to update the value
+        of a setting without destroying the description.
+        """
+        # make sure settings are always associated to the root command (which derives from GenericCommand)
+        if "GenericCommand" not in [x.__name__ for x in self.__class__.__bases__]:
+            return
+        key = self.__get_setting_name(name)
+        __config__[key][0] = value
```

Comment:
I would think we'd need `global __config__`

Reply:
this code is the exact code from `add_setting`